File: JavaScript/739-Daily Temperatures.js
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function(temperatures) {
    let res = new Array(temperatures.length).fill(0);
    let stack = new Array();
    for (let i=0; i<temperatures.length; i++){
        while (stack.length && temperatures[i] > temperatures[stack[stack.length-1]] ){
            let prevIdx = stack[stack.length-1];
            res[prevIdx] = i - prevIdx;
            stack.pop();
        }
        stack.push(i);
    }
    return res;
};

File: C++/739-Daily Temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        stack<int> stk;
        int N = temperatures.size();
        vector<int> res(N, 0);
        for (int i=0; i<temperatures.size(); i++) {
            while (!stk.empty() && temperatures[stk.top()] < temperatures[i]) {
                int ind = stk.top();
                stk.pop();
                res[ind] = i-ind;
            }
            stk.push(i);
        }
        return res;
    }
};